src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCopyFileFromContainer.groovy

```diff
@@ -15,55 +15,209 @@
  */
 package com.bmuschko.gradle.docker.tasks.container
 
+import groovy.io.FileType
+import org.gradle.api.GradleException
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.Optional
-import org.gradle.api.tasks.OutputFile
-
-import java.util.zip.GZIPOutputStream
 
 class DockerCopyFileFromContainer extends DockerExistingContainer {
+    /**
+     * Path inside container
+     */
     @Input
-    String resource
+    String remotePath
 
+    /**
+     * Path on host to write remotePath to or into.
+     *
+     * If hostPath does not exist it will be created relative to
+     * what we need it to be (e.g. regular file or directory).
+     * This is consistent with how 'docker cp' behaves.
+     */
     @Input
-    @OutputFile
-    File destFile
+    @Optional
+    String hostPath
 
+    /**
+     * Whether to leave file in its compressed state or not.
+     *
+     * Docker CP command hands back a tar stream regardless if we asked
+     * for a regular file or directory. Thus, we can give the caller
+     * back the tar file as-is or explode it to some destination like
+     * 'docker cp' does.
+     */
     @Input
     @Optional
     Boolean compressed = Boolean.FALSE
 
     @Override
     void runRemoteCommand(dockerClient) {
-        logger.quiet "Copying '${getResource()}' from container with ID '${getContainerId()}' to '${getDestFile()}'."
 
-        getDestFile().withOutputStream  { OutputStream out ->
-            def input
+        if (!hostPath)
+            hostPath = "${project.projectDir}"
```

Comment:
Should this be `project.projectDir.path` instead? We could simply assign this as default value to the property `hostPath` and make it non-optional.